### PR TITLE
fix: Rename RC to DevServer

### PR DESCRIPTION
**Pull request by jeromelaban**
## PR Type

What kind of change does this PR introduce?
- Feature

## What is the new behavior?

Rename RC server to DevServer.

### <samp>🤖 Generated by Copilot at d8e6bae</samp>

This pull request renames and reorganizes the packages and files related to the development server feature for Uno Platform projects. It replaces the `Uno.UI.RemoteControl` package and its subpackages with the `Uno.UI.DevServer` package and its subpackage `Uno.UI.DevServer.Messaging`. It also updates the nuspec files, project files, and targets files to reflect the new names and versions. The old package names are deprecated and redirect to the new ones.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by ajpinedam — APPROVED** [review]
DevServer is way more clear on what it does. Great change!

File: build/nuget/Uno.WinUI.DevServer.nuspec [hunk 1]
```diff
@@ -0,0 +1,213 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
+	<metadata minClientVersion="5.0">
+		<id>Uno.UI.RemoteControl</id>
+		<version>0.1.0.0</version>
+		<title>Uno.UI.RemoteControl</title>
```
Comment (Youssef1313) [inline build/nuget/Uno.WinUI.DevServer.nuspec:6]: Should these be DevServer as well?

File: build/nuget/Uno.WinUI.DevServer.nuspec [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+		<description>Development Server for Uno Platform projects, such as XAML Hot Reload.</description>
+		<copyright>Copyright (C) 2015-2023 Uno Platform Inc. - all rights reserved</copyright>
+		<repository type="git" url="https://github.com/unoplatform/uno.git" branch="$branch$" commit="$commitid$" />
+
+		<dependencies>
+
+			<!-- net8.0-ios -->
+			<group targetFramework="net8.0-ios">
+				<dependency id="Uno.WinUI" version="1.29.0-dev.93" />
+				<dependency id="Uno.WinUI.RemoteControl.Messaging" version="1.29.0-dev.93" />
```
Comment (Youssef1313) [inline build/nuget/Uno.WinUI.DevServer.nuspec:22]: DevServer here as well?

File: build/nuget/Uno.WinUI.DevServer.nuspec (hunk 1 above)
Comment (jeromelaban) [reply build/nuget/Uno.WinUI.DevServer.nuspec:6]: Indeed!

File: build/nuget/Uno.WinUI.DevServer.nuspec (hunk 2 above)
Comment (jeromelaban) [reply build/nuget/Uno.WinUI.DevServer.nuspec:22]: Same indeed!

**Comment by MartinZikmund** [comment]
I'm a bit afraid of confusion with Wasm.Bootstrap.DevServer 🤔

**Comment by jeromelaban** [comment]
> I'm a bit afraid of confusion with Wasm.Bootstrap.DevServer 🤔

That's fair. Still, the package is different yet the objective is the same, helping during development.

File: build/Uno.UI.Build.csproj [hunk 3]
```diff
@@ -56,7 +56,8 @@
 		<ItemGroup>
 			<_PackageToUpdate Include="Uno.UI" />
 			<_PackageToUpdate Include="Uno.UI.RemoteControl" />
-			<_PackageToUpdate Include="Uno.UI.RemoteControl.Messaging" />
+			<_PackageToUpdate Include="Uno.UI.DevServer" />
+			<_PackageToUpdate Include="Uno.UI.DevServer.Messaging" />
```
Comment (Youssef1313) [inline build/Uno.UI.Build.csproj:60]: I don't see any usage of `_PackageToUpdate`.

File: build/nuget/Uno.WinUI.DevServer.nuspec [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+			<!-- .NET 7 Android -->
+			<group targetFramework="net7.0-android">
+				<reference file="Uno.UI.RemoteControl.dll" />
+			</group>
+
+			<!-- .NET 7 iOS -->
+			<group targetFramework="net7.0-ios">
+				<reference file="Uno.UI.RemoteControl.dll" />
+			</group>
```
Comment (Youssef1313) [inline build/nuget/Uno.WinUI.DevServer.nuspec:141]: Don't we need `macos` and `maccatalyst` here?

File: build/Uno.UI.Build.csproj (hunk 3 above)
Comment (jeromelaban) [reply build/Uno.UI.Build.csproj:60]: Indeed we can remove.

File: build/nuget/Uno.WinUI.DevServer.nuspec (hunk 4 above)
Comment (jeromelaban) [reply build/nuget/Uno.WinUI.DevServer.nuspec:141]: That whole block is not useful, actually. We can let nuget do the references automatically.

**Pull request merged**